tests/server/api/region_test.go

```diff
@@ -0,0 +1,287 @@
+
+	err := tu.CheckPostJSON(testDialClient, fmt.Sprintf("%s/regions/accelerate-schedule/batch", urlPrefix), []byte(body), tu.StatusOK(re))
+	suite.NoError(err)
+	idList := leader.GetRaftCluster().GetSuspectRegions()
+	if sche := cluster.GetSchedulingPrimaryServer(); sche != nil {
+		idList = sche.GetCluster().GetCoordinator().GetCheckerController().GetSuspectRegions()
+	}
+	suite.Len(idList, 4)
+}
+
+func (suite *regionTestSuite) TestScatterRegions() {
+	env := tests.NewSchedulingTestEnvironment(suite.T())
+	env.RunTestInTwoModes(suite.checkScatterRegions)
+}
+
+func (suite *regionTestSuite) checkScatterRegions(cluster *tests.TestCluster) {
+	leader := cluster.GetLeaderServer()
+	urlPrefix := leader.GetAddr() + "/pd/api/v1"
+	re := suite.Require()
+	r1 := core.NewTestRegionInfo(601, 13, []byte("b1"), []byte("b2"))
+	r1.GetMeta().Peers = append(r1.GetMeta().Peers, &metapb.Peer{Id: 5, StoreId: 14}, &metapb.Peer{Id: 6, StoreId: 15})
+	r2 := core.NewTestRegionInfo(602, 13, []byte("b2"), []byte("b3"))
+	r2.GetMeta().Peers = append(r2.GetMeta().Peers, &metapb.Peer{Id: 7, StoreId: 14}, &metapb.Peer{Id: 8, StoreId: 15})
+	r3 := core.NewTestRegionInfo(603, 13, []byte("b4"), []byte("b4"))
+	r3.GetMeta().Peers = append(r3.GetMeta().Peers, &metapb.Peer{Id: 9, StoreId: 14}, &metapb.Peer{Id: 10, StoreId: 15})
+	tests.MustPutRegionInfo(re, cluster, r1)
+	tests.MustPutRegionInfo(re, cluster, r2)
+	tests.MustPutRegionInfo(re, cluster, r3)
+	for i := 13; i <= 16; i++ {
+		s1 := &metapb.Store{
+			Id:        uint64(i),
+			State:     metapb.StoreState_Up,
+			NodeState: metapb.NodeState_Serving,
+		}
+		tests.MustPutStore(re, cluster, s1)
+	}
+	body := fmt.Sprintf(`{"start_key":"%s", "end_key": "%s"}`, hex.EncodeToString([]byte("b1")), hex.EncodeToString([]byte("b3")))
+
+	err := tu.CheckPostJSON(testDialClient, fmt.Sprintf("%s/regions/scatter", urlPrefix), []byte(body), tu.StatusOK(re))
+	suite.NoError(err)
+	oc := leader.GetRaftCluster().GetOperatorController()
+	if sche := cluster.GetSchedulingPrimaryServer(); sche != nil {
+		oc = sche.GetCoordinator().GetOperatorController()
+	}
+
+	op1 := oc.GetOperator(601)
+	op2 := oc.GetOperator(602)
+	op3 := oc.GetOperator(603)
+	// At least one operator used to scatter region
+	suite.True(op1 != nil || op2 != nil || op3 != nil)
+
+	body = `{"regions_id": [601, 602, 603]}`
+	err = tu.CheckPostJSON(testDialClient, fmt.Sprintf("%s/regions/scatter", urlPrefix), []byte(body), tu.StatusOK(re))
+	suite.NoError(err)
+}
+
+func (suite *regionTestSuite) TestCheckRegionsReplicated() {
+	env := tests.NewSchedulingTestEnvironment(suite.T(),
+		func(conf *config.Config, serverName string) {
+			conf.Replication.EnablePlacementRules = true
+		})
+	// FIXME: enable this test in two modes.
+	env.RunTestInPDMode(suite.checkRegionsReplicated)
```

Comment:
This seem to be a new bug. BTW, we need to fix below test.
![image](https://github.com/tikv/pd/assets/19542290/3bd00f7b-57ea-4448-a990-660cf6b32a16)

Reply:
it always try to add rule peer, maybe there is a problem in placement rule sync

Reply:
I will do it in another pr https://github.com/tikv/pd/issues/7418